bigquery/google/cloud/bigquery/client.py

```diff
@@ -492,25 +492,45 @@ def load_table_from_storage(self, job_id, destination, *source_uris):
         """
         return LoadJob(job_id, destination, source_uris, client=self)
 
-    def copy_table(self, job_id, destination, *sources):
-        """Construct a job for copying one or more tables into another table.
+    def copy_table(self, sources, destination, **kwargs):
+        """Start a job for copying one or more tables into another table.
 
         See
         https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#configuration.copy
 
-        :type job_id: str
-        :param job_id: Name of the job.
+        :type sources: One of:
+                       :class:`~google.cloud.bigquery.table.TableReference`
+                       sequence of
+                         :class:`~google.cloud.bigquery.table.TableReference`
+        :param sources: Table or tables to be copied.
 
-        :type destination: :class:`google.cloud.bigquery.table.Table`
+
+        :type destination: :class:`google.cloud.bigquery.table.TableReference`
         :param destination: Table into which data is to be copied.
 
-        :type sources: sequence of :class:`google.cloud.bigquery.table.Table`
-        :param sources: tables to be copied.
+        :type kwargs: dict
+        :param kwargs: Additional keyword arguments.
+
+        :Keyword Arguments:
+            * *job_config*
+              (:class:`google.cloud.bigquery.job.CopyJobConfig`) --
+              (Optional) Extra configuration options for the copy job.
+            * *job_id* (``str``) --
+              Additional content
+              (Optional) The ID of the job.
 
         :rtype: :class:`google.cloud.bigquery.job.CopyJob`
         :returns: a new ``CopyJob`` instance
         """
-        return CopyJob(job_id, destination, sources, client=self)
+        job_config = kwargs.get('job_config')
+        job_id = _make_job_id(kwargs.get('job_id'))
+
+        if isinstance(sources, TableReference):
```

Comment:
I might be more inclined to do this one the converse way:

```python
if not isinstance(sources, collections.Sequence):
    sources = [sources]
```

My logic is that if at some future point we accept some additional input (say, for example, a dictionary that maps nicely to a `TableReference`, this logic will continue to work.

That may not matter here. I would care a lot if this was mucking with an autogen `_pb2` type, but it is not doing that.

So, essentially, dropping the thought here, but I trust your judgment.

Reply:
tbh, I wasn't aware how to test for being a sequence in modern Python. I'll use your code.